### PR TITLE
Amls 4702 - Converted connectors to dependency injection.

### PR DESCRIPTION
**Pull request by slavvco**
In this PR connectors have been converted to dependency injection. Changes in them forced another changes what is reflected in PR. Conversion of services was necessary to complete this change, so #931 AMLS-4703 was merged in this branch. Changes to connectors forced the changes to various unit tests throughout the service as well - some of them are cosmetic (but still important) like for example changing: extends PlaySpec with MockitoSugar  with OneAppPerSuite to AmlsSpec only.
Some of unit tests required to bind KeystoreConnector to its mock when creating app instance. All below...

## Related / Dependant PRs?

#931 AMLS-4703 need to be merged before this PR.

## How Has This Been Tested?

- unit test run
- regression pack run

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: release_notes/AMLS-4702.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+ + [AMLS-4702](https://jira.tools.tax.service.gov.uk/browse/AMLS-4702) - 'Converted connectors to use dependency injection'
```
Comment (matthewharris1) [inline release_notes/AMLS-4702.txt]: Does this need release notes for services DI changes aswell as connectors as they've been merged together?

File: test/controllers/responsiblepeople/TimeAtAdditionalExtraAddressControllerSpec.scala [hunk 2]
```diff
@@ -204,7 +204,9 @@ class TimeAtAdditionalExtraAddressControllerSpec extends AmlsSpec with MockitoSu
     }
   }
   "App must start" in {
-    val app = GuiceApplicationBuilder().build()
+    val app = GuiceApplicationBuilder()
```
Comment (matthewharris1) [inline test/controllers/responsiblepeople/TimeAtAdditionalExtraAddressControllerSpec.scala]: Did we say we were keeping these specs or removing them? Not sure if the keystoreConnector bind has something to do with it but It seems pointless keeping tests to see if the app can start.

File: test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala [hunk 3]
```diff
@@ -170,7 +171,9 @@ class VATRegisteredControllerSpec extends AmlsSpec with MockitoSugar with ScalaF
       }
     }
     "App" in {
-      val app = GuiceApplicationBuilder().build()
+      val app = GuiceApplicationBuilder()
```
Comment (matthewharris1) [inline test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala]: ^

File: release_notes/AMLS-4702.txt (hunk 1 above)
Comment (slavvco) [reply release_notes/AMLS-4702.txt]: that's release note for my changes only and here I don't need the one for services, as services need to be merged first to master with it's own release note
I've merged services here just to be sure that my changes are working together with changes from branch 4703

File: test/controllers/responsiblepeople/TimeAtAdditionalExtraAddressControllerSpec.scala (hunk 2 above)
Comment (slavvco) [reply test/controllers/responsiblepeople/TimeAtAdditionalExtraAddressControllerSpec.scala]: I've never heard about removing those specs, but in order to run this spec, bind is needed

Comment (slavvco) [reply test/controllers/responsiblepeople/TimeAtAdditionalExtraAddressControllerSpec.scala]: but I could :)

File: test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala (hunk 3 above)
Comment (slavvco) [reply test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala]: same as above

Comment (slavvco) [reply test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala]: anyway I'll check some specs again

Comment (matthewharris1) [reply test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala]: Okay mate, just let me know once thats done and the rest looks good. Just seems pointless to me adding additional unit tests just to see whether or not the app runs. That should just be a given. I can see why its beneficial when amending the connectors to save you having to constantly check whether the page loads.

Comment (slavvco) [reply test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala]: you know tests above are not mine, looks like they are yours haha :P I was just updating some failing ones, seems I've added some unnecessary bindings, but I'm fixing that

File: test/controllers/responsiblepeople/TimeAtAdditionalExtraAddressControllerSpec.scala (hunk 2 above)
Comment (slavvco) [reply test/controllers/responsiblepeople/TimeAtAdditionalExtraAddressControllerSpec.scala]: done

File: test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala (hunk 3 above)
Comment (slavvco) [reply test/controllers/responsiblepeople/VATRegisteredControllerSpec.scala]: done

**Comment by matthewharris1** [comment]
Looks good to me, might want another devs opinion because theres a large amount of files changed :+1:

**Comment by slavvco** [comment]
thanks guys, after services DI is merged into master I'll update this branch, then will be ready for AC

**Comment by slavvco** [comment]
after pulling master no conflicts

File: app/connectors/GovernmentGatewayConnector.scala [hunk 4]
```diff
@@ -17,22 +17,25 @@
...
-trait GovernmentGatewayConnector {
+class GovernmentGatewayConnector @Inject()(val http: WSHttp,
+                                           val appConfig: AppConfig) {
 
-  protected[connectors] def http: CorePost
-  protected def enrolUrl: String
-  private[connectors] def audit: Audit
+  val audit: Audit = new Audit(AuditHelper.appName, AMLSAuditConnector)
+  protected def enrolUrl = appConfig.enrolUrl
+  private[connectors] val duplicateEnrolmentMessage = "The service HMRC-MLR-ORG requires unique identifiers"
```
Comment (ajchilts) [inline app/connectors/GovernmentGatewayConnector.scala]: Should these strings be in the 'Messages' file or do we not have access to that here?

**Comment by matthewharris1** [comment]
Acceptance test suite passing :+1: Merge when you want

File: app/connectors/GovernmentGatewayConnector.scala (hunk 4 above)
Comment (slavvco) [reply app/connectors/GovernmentGatewayConnector.scala]: looks like messages api doesn't like to work in this case, so I'd rather stay with current solution if you don't mind

**Pull request merged**